src/web/user_agent.rs

```diff
@@ -28,6 +68,23 @@ pub fn strip_ua(ua: &str) -> String {
     )
 }
 
+pub fn get_device_info(ua: &str) -> (OsFamily, FormFactor) {
+    let WootheeResult { os, .. } = Parser::new().parse(ua).unwrap_or_default();
+
+    let os_family = match os {
```

Comment:
I doubt it but sure. I was expecting we'd need to drop woothee because its dataset isn't looking as good as [ua-parser](https://github.com/ua-parser)'s (which now has a rust impl) -- but I ended up fleshing this out more and it's probably good enough for our usage.